### PR TITLE
XGQ hw emu coverity fixes

### PR DESCRIPTION
**Pull request by larry9523**
Fixes a few uninitialized variable coverity issue and a memory leak.

### CODE REVIEW

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.cxx [hunk 1]
```diff
@@ -200,6 +201,7 @@ namespace hwemu {
   xgq_cmd::xgq_cmd()
   {
     cmdid = ++next_uid; //! Assign unique ID for each new command
+    ert_pkt = NULL;
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.cxx:204]: nullptr or dinged by clangtidy

**Review by stsoe — COMMENTED** [review]
It's unfortunate that xgd_hwemu.h also does not compile stand-alone.
This is a c++ header, and uses many types not included.

I am adding clangtidy review to our build process and that is how I am catching these errors, I am using this PR as a test.   If you can fix the missing includes it would be great.  Thanks!

In order to compile xgd_hwemu.h stand-alone, I used this command (fix the paths) if you want to try.   Else just forget and we can fix later.

% g++ -std=c++14 -c     -I/home/soeren/git/stsoe/XRT.tmp/src/runtime_src     -I/home/soeren/git/stsoe/XRT.tmp/src/runtime_src/core/include     -I/home/soeren/git/stsoe/XRT.tmp/src/runtime_src/core/pcie/emulation/common_em     -I/home/soeren/git/stsoe/XRT.tmp/build/Review/runtime_src/core/pcie/emulation/common_em     -o /tmp/compileme.o src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.h

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.cxx (hunk 1 above)
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.cxx:204]: warning: use nullptr [modernize-use-nullptr]

```cpp
ert_pkt = NULL;
          ^~~~
          nullptr
```

**Review by stsoe — COMMENTED** [review]
clang-tidy made some suggestions

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.cxx (hunk 1 above)
Comment (larry9523) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.cxx:204]: I knew this one. I was trying to follow 
 ert_pkt = NULL;
in https://github.com/Xilinx/XRT/blob/master/src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/mbscheduler_hwemu.cxx#L41

But I am fixing it in xgq_hwemu now. Thanks.

**Comment by larry9523** [comment]
> It's unfortunate that xgd_hwemu.h also does not compile stand-alone.
> This is a c++ header, and uses many types not included.
> 
> I am adding clangtidy review to our build process and that is how I am catching these errors, I am using this PR as a test. If you can fix the missing includes it would be great. Thanks!
> 
> In order to compile xgd_hwemu.h stand-alone, I used this command (fix the paths) if you want to try. Else just forget and we can fix later.
> 
> % g++ -std=c++14 -c -I/home/soeren/git/stsoe/XRT.tmp/src/runtime_src -I/home/soeren/git/stsoe/XRT.tmp/src/runtime_src/core/include -I/home/soeren/git/stsoe/XRT.tmp/src/runtime_src/core/pcie/emulation/common_em -I/home/soeren/git/stsoe/XRT.tmp/build/Review/runtime_src/core/pcie/emulation/common_em -o /tmp/compileme.o src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.h

Thanks Soren. I just pushed a fix.

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.h [hunk 2]
```diff
@@ -35,12 +35,19 @@
...
 #include <boost/pool/object_pool.hpp>
+#include <condition_variable>
+#include <cstdint>
+#include <list>
+#include <mutex>
+#include <thread>
+
+#include "em_defines.h"
+#include "ert.h"
+#include "xgq.h"
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.h:50]: Last missing header is vector I think.

```
/home/soeren/git/stsoe/XRT.tmp/src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.h:156:12: error: no template named 'vector' in namespace 'std' [clang-diagnostic-error]
      std::vector<uint32_t> sq_buf;
```

Comment (larry9523) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/xgq_hwemu.h:51]: Oh, sorry. I was using the compile command you suggested and it did not show any error. 

I just added the vector header.

**Pull request merged**